### ISSUE TITLE
Find some columns, but no dump or enumeration of them

### ISSUE THREAD

**Issue by strano666** [sqlmapproject/sqlmap#4002]
I can find some columns via `--search` option, but if try to enumerate the columns of the specific table, these columns there are not there. Also if i try a dump of the entire table.
But if i try a dump of single "hidden" column, it's ok

search a `LIKE pass` column in a table, with success
```
python sqlmap.py -u '<URL>' --level=3 --risk=3 --random-agent -D <DB_NAME> -T <DB_TABLE> --search -C pass
[.....]
columns LIKE 'pass' were found in the following databases:
Database: <DB_NAME>
Table: <DB_TABLE>
[1 column]
+--------+-------------+
| Column | Type        |
+--------+-------------+
| pass   | varchar(50) |
+--------+-------------+
```
and then enumerate the columns of the same table, but i can not see this column
```
python sqlmap.py -u '<URL>' --level=3 --risk=3 --random-agent -D <DB_NAME> -T <DB_TABLE> --columns
[....]
[17:08:20] [INFO] fetching columns for table '<DB_TABLE>' in database '<DB_NAME>'
[17:08:20] [INFO] used SQL query returns 70 entries
[17:08:20] [INFO] heuristics detected web page charset 'ascii'
Database: <DB_NAME>
Table: <DB_TABLE>
[13 columns]
+-------------+--------+
| Column      | Type   |
+-------------+--------+
| col1        | text   |
| col2       | double |
| col3      | double |
| col4     | text   |
| col5     | text   |
+-------------+--------+
```
But if i try to dump the single entry of the column, it is ok
```
python sqlmap.py -u '<URL>' --level=3 --risk=3 --random-agent -D <DB_NAME> -T <DB_TABLE> -C pass --dump --start=1 --stop=2
[.....]
web server operating system: Linux Debian 8.0 (jessie)
web application technology: Apache 2.4.10
back-end DBMS: MySQL >= 5.1
Database: <DB_NAME>
Table: <DB_TABLE>
[3 entries]
+-----------------+
| pass            |
+-----------------+
| <ENTRY>         |
|
|
+-----------------+
```

**Expected behavior**
A coherent behaviour between searching, enumeration and dumping

**Screenshots**
Enumerating columns
<img width="738" alt="enumerate_columns" src="https://user-images.githubusercontent.com/4613342/68409809-0936fc80-0188-11ea-9aa6-26739c64cbe8.png">

Searching columns
<img width="845" alt="Schermata 2019-11-07 alle 17 58 43" src="https://user-images.githubusercontent.com/4613342/68410165-94b08d80-0188-11ea-9918-c337ccbf4ccf.png">

**Running environment:**
 - sqlmap version: 1.3.10.22#dev
 - Installation method git
 - Operating system: macOS Mojave 10.14.4
 - Python version 2.7.16

**Target details:**
 - DBMS: MySQL
 - SQLi techniques found by sqlmap: boolean-based blind, error-based, time-based blind, UNION query

**Additional context**
It happens also with others column name

**Comment by stamparm** [comment]
1) Sorry, but your english is poor here. For example: `but if try to enumerate the columns of the specific table, these columns there are not there` <- I don't know what's the meaning of this
2) I am missing things here. This kind of behavior is not normal. As you can suspect, I am not able to reproduce. This is an example "alike" run against the (MySQL) test site:

```
$ python sqlmap.py -u "http://testphp.vulnweb.com/artists.php?artist=1" --search -D acuart -T artists -C id
...
[22:50:42] [INFO] searching columns LIKE 'id' for table 'artists' in database 'acuart'
[22:50:42] [INFO] fetching columns LIKE 'id' for table 'artists' in database 'acuart'
[22:50:42] [INFO] used SQL query returns 1 entry
columns LIKE 'id' were found in the following databases:
Database: acuart
Table: artists
[1 column]
+-----------+--------+
| Column    | Type   |
+-----------+--------+
| artist_id | int(5) |
+-----------+--------+
```

```
$ python sqlmap.py -u "http://testphp.vulnweb.com/artists.php?artist=1" --dump -D acuart -T artists -C artist_id --start=1 --stop=2
...
Database: acuart                                                                                                                                                                                             
Table: artists
[2 entries]
+-----------+
| artist_id |
+-----------+
| 1         |
| 2         |
+-----------+
```

```
$ python sqlmap.py -u "http://testphp.vulnweb.com/artists.php?artist=1" -D acuart -T artists --columns
...
Database: acuart                                                                                                                                                                                             
Table: artists
[3 columns]
+-----------+-------------+
| Column    | Type        |
+-----------+-------------+
| adesc     | text        |
| aname     | varchar(50) |
| artist_id | int(5)      |
+-----------+-------------+
```

**Comment by strano666** [comment]
Sorry for my english.
I wrote in a rush, my fault. Anyway i've fixed (i hope) :)

TL;DR:

1.  I find the column `C` in `T` table of `D` database (`--search` flag) 
2.  I enumerate the columns of `T`, but there is not `C`
3.  If i dump only `C` column, i can retrieve the values
4.  If i dump the entire table, i can retrieve only the values of the columns enumerated at point 2.
